nasa/earthdata-search
# EDSC-3923: Add clientId to EDD startDownload link

# Overview

### What is the feature?

Add the `clientId` parameter to the `startDownload` link to track which EDSC environment initializes  the download

### What is the Solution?

Use the `env` property from the Application config for the `clientId` parameter and place it in the EDD URL.

### What areas of the application does this impact?

EDD link generation.

# Testing

- Select some granules to download
- Choose the Direct Download access method
- Click the Download Data button
- Once the file URLs populate in the status window, click the Download Files button
- Close the Earthdata Download application if it opens, or refuse your browser's request to open it
- Right-click on the Open Earthdata Download button, and copy the URL
- Paste the URL somewhere, and verify that the it contains `&clientId=dev` after the mocked `authToken`

![image](https://github.com/nasa/earthdata-search/assets/10775395/f9294d7f-9bc0-420b-a77e-4d7c994e390b)

# Checklist
